<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">

<html>
<head>
		<meta http-equiv="content-type" content="text/html;charset=utf-8">
		<title>Integrative Biology 200 - Syllabus and Handouts</title>

        <link href="assets/style.css" rel="stylesheet">
        <link href="assets/bootstrap.css" rel="stylesheet">
</head>


<body>


    <!-- Fixed navbar -->
    <nav class="navbar navbar-default navbar-fixed-top">
      <div class="container">
        <div class="navbar-header">
          <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false" aria-controls="navbar">
            <span class="sr-only">Toggle navigation</span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
          </button>
        </div>
        <div id="navbar" class="navbar-collapse collapse">
          <ul class="nav navbar-nav">
            <li><a href="index.html">Home</a></li>
            <li class="active"><a href="syllabus.html">Syllabus</a></li>
            <li><a href="readings.html">Recommended Readings</a></li>
            <li><a href="quizzes.html">Past Quizzes</a></li>
            <li><a href="links.html">Useful Links</a></li>
         </div><!--/.nav-collapse -->
      </div>
    </nav>


<!-- Body Text -->
<br><br>
<h1>Integrative Biology 200<br/>
<small>Syllabus and Handouts</small></h1>

<br><br>
Download syllabus as <a href="IB200_syllabus_2016.pdf">pdf</a>.<br>
<table class="table table-bordered table-striped">
<thead>
<tr>
  <td valign="top"><b>Date</b> </td>
  <td valign="top"><b>Lecture</b> </td>
  <td valign="top"><strong>Required Readings<br>
        (see also <a href="readings.html">Recommended Readings</a>)</strong></td>
  <td valign="top"><b>Lab</b> </td>
</tr>
</thead>

<tr>
  <td>Wednesday,<br>Jan. 20</td>
  <td>Introduction - contemporary issues in phylogenetic systematics - what is at stake? (KWW)</td>
  <td><i>Tree Thinking</i>: Chap. 1</td>
  <td>LAB 01:  discussion: student interests; get acquainted roundtable; Tour of systematics collections, labs, 
      and resources in VLSB
      </td>
</tr>   

<tr>
  <td>Friday,<br>Jan. 22</td>
  <td>Introduction - contemporary issues in comparative methods (DDA)</td>
  <td>Coddington, J. A. 1994. The roles of homology and convergence in studies of adaptation. Academic Press, London.<br><br>
  Ackerly, D.D. 2009. Phylogenetics and comparative methods. Pages 117-125 in 
  Levin et al., eds. <i>The Princeton guide to ecology.</i> Princeton University Press, 2009.</td>
  <td></td>
</tr>

<!-- week -->
<tr>
  <td>Monday,<br>Jan. 25</td>
  <td>History & philosophy of phylogenetics; the Hennig Principle: homology; synapomorphy (BDM) 
  <br><br>
  <a href="lect/lect03.pdf">Lecture Notes</a></td>
  <td>
  Required reading: <i>Tree Thinking</i>: Chap. 2<br><br>
  
  Supplemental: <a href="readings/Mishler 2009 (Taxon).pdf">B.D. Mishler. 2009. Three centuries of paradigm changes in biological
  classification: is the end in sight? Taxon 58: 61-67.</a>
 <br><br> 
  <a href="readings/Mishler 2014 (History).pdf">B.D. Mishler. 2014. History and theory in the development of phylogenetics in botany. In A. 
  Hamilton (ed.), The Evolution of Phylogenetic Systematics, pp. 189-210. UC Press.</a></td>
  <td></td>
</tr>

<tr>
  <td>Wednesday,<br>Jan. 27</td>
  <td>Morphological data I: Character analysis; What is a data matrix?
  Homoplasy; Ontogeny & structure of plants & animals; The role of morphology (BDM)
  <br><br>
  <a href="lect/lect04.pdf">Lecture Notes</a></td>
  <td><a href="readings/Mishler 2005.pdf">B.D. Mishler.  2005.  The logic of the data matrix in phylogenetic analysis. In V.A. Albert (ed.), Parsimony, Phylogeny, and Genomics, pp. 57-70.  Oxford University Press.</a></td>
  <td><a href="labs/02/lab02.pdf">LAB 02: Discussion about homology; Introduction to Nexus and Newick files; Introduction to FigTree and Mesquite<br><br>
  Please do Part 1 before coming to lab (installing software).</a></td>
</tr>

<tr>
  <td>Friday,<br>Jan. 29</td>
  <td>Morphological data II: Character coding [primary homology, polarity, additivity, etc.]; (KWW)
  <br><br>
  <a href="lect/lect05.pdf">Lecture Notes</a></td>
  <td><i>Tree Thinking</i>: pp 77-95, 399-403
  <br><br>
  <a href="http://link.springer.com/content/pdf/10.1007/978-3-642-39979-4_6.pdf">
  Rieppel, O. (2015). Homology: A Philosophical and Biological Perspective. Handbook of Paleoanthropology, 295-315.</a>
  </td>
  <td></td>
</tr>

<!-- week -->
<tr>
  <td>Monday,<br>Feb. 1</td>
  <td>Molecular data I: General introduction; types of molecular data (DNA hybridization; allozymes; restriction sites,     DNA sequences, ESTs; comparative genomics) (BDM)
  <br><br>
  <a href="lect/lect06.pdf">Lecture Notes</a></td>
  <td><a href="https://mesquiteproject.wikispaces.com/page/diff/Analyzing+Molecular+Data/519996188">
  Maddison, W. P. and D.R. Maddison. 2011. Mesquite: a modular system for evolutionary analysis. Version 2.75 Chapter on Molecular Data</a></td>
  <td></td>
</tr>

<tr>
  <td>Wednesday,<br>Feb. 3</td>
  <td>Molecular data II: Sequence alignment (KWW)
  <br><br>
  <a href="lect/lect07.pdf">Lecture Notes</a></td>
  <td><i>Tree Thinking</i>: pp 86-89, 195-200</td>
  <td><b>PROJECT TOPIC DUE -- discuss in class</b><br><br>
    <a href="labs/03/lab03.pdf">LAB 03: Introduction to GenBank, BLAST, and FASTA files; 
    sequence analysis and alignment (Clustal, Muscle, AliView)</a>
    </td>
</tr>

<tr>
  <td>Friday,<br>Feb. 5</td>
  <td>Phylogenetic trees I:  reconstruction; models, algorithms & assumptions (BDM)
  <br><br>
  <a href="lect/lect08.pdf">Lecture Notes</a></td>
  <td><i>Tree Thinking</i>: pp 35-53</td>
  <td></td>
</tr>

<!-- week -->
<tr>
  <td>Monday,<br>Feb. 8</td>
  <td>Phylogenetic trees II: Phenetics; distance-based algorithms (KWW)
  <br><br>
  <a href="lect/lect09.pdf">Lecture Notes</a></td>
  <td><i>Tree Thinking</i>: pp 231-238</td>
  <td></td>
</tr>

<tr>
  <td>Wednesday,<br>Feb. 10</td>
  <td>Phylogenetic trees III: Parsimony; Measures of support and robustness (KWW)
  <br><br>
  <a href="lect/lect10.pdf">Lecture Notes</a></td>
  <td><i>Tree Thinking</i>: pp 95-106, 173-215, 271-284
  <br><br>
  <a href="http://www.pagepress.org/journals/index.php/eb/article/view/eb.2010.e7/2536">
  Anisimova, M., Cannarozzi, G., & Liberles, D. A. (2010). Finding the balance between the mathematical and biological optima in multiple sequence alignment. Trends in Evolutionary Biology, 2(1), 7.</a>
  <br><br>
  <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1096-0031.2003.tb00311.x/abstract">
  Grant, T., & Kluge, A. G. (2003). Data exploration in phylogenetic inference: scientific, heuristic, or neither. Cladistics, 19(5), 379-418.</a>
  </td>
  <td><a href="labs/04/lab04.pdf">LAB 04: Distance and parsimony inference using PAUP*; UPGMA, neighbor-joining, bootstrap, and jackknife</a>
    </td>
</tr>

<tr>
  <td>Friday,<br>Feb. 12</td>
  <td>Phylogenetic trees IV:  Maximum likelihood; molecular evolution and phylogenetics (KWW)
  <br><br>
  <a href="lect/lect11.pdf">Lecture Notes</a></td>
  <td><i>Tree Thinking</i>: pp 217-231, 238-247
  <br><br>
  <a href="http://www.jstor.org/stable/3070925?seq=1#page_scan_tab_contents">
  De Queiroz, K., & Poe, S. (2001). Philosophy and phylogenetic inference: a comparison of likelihood and parsimony methods in the context of Karl Popper's writings on corroboration. Systematic Biology, 50(3), 305-321.</a>
  </td>
  <td></td>
</tr>

<!-- week -->
<tr>
  <td>Monday,<br>Feb. 15</td>
  <td colspan=3><i>Academic and Administrative Holiday - NO CLASS</i></td>
</tr>

<tr>
  <td>Wednesday,<br>Feb.17</td>
  <td>Phylogenetic trees V: Bayesian methods and Markov chain Monte Carlo (WF)
  <br><br>
  <a href="lect/lect12.pdf">Lecture Notes</a></td>
  <td><i>Tree Thinking</i>: pp 247-258<br><br>
  Huelsenbeck, J. P., Larget, B., Miller, R. E., & Ronquist, F. 2002. Potential applications and pitfalls of Bayesian inference of phylogeny. Systematic Biology, 51(5), 673–88. <a href="http://doi.org/10.1080/10635150290102366">http://doi.org/10.1080/10635150290102366</a><br><br>
  Holder, M., & Lewis, P. O. 2003. Phylogeny estimation: traditional and Bayesian approaches. Nature Reviews. Genetics, 4(4), 275–84. <a href="http://doi.org/10.1038/nrg1044">http://doi.org/10.1038/nrg1044</a></td>
  <td><a href="labs/05/lab05.pdf">LAB 05: Maximum likelihood inference; models of DNA sequence evolution; jModelTest and PAUP*; RAxML and CIPRES supercomputer web 
    interface</a>
    </td>
</tr>

<tr>
  <td>Friday,<br>Feb. 19</td>
  <td>Phylogenetic trees VI: Dating in the 21st century: clocks, & calibrations; proper use of fossils (KWW)
  <br><br>
  <a href="lect/lect13.pdf">Lecture Notes</a></td>
  <td><i>Tree Thinking</i>: pp 53-58
  <br><br>
  <a href="http://sysbio.oxfordjournals.org/content/61/2/346.full">
  Parham et al. (2011). Best practices for justifying fossil calibrations. Systematic Biology, syr107.</a></td>
  <td></td>
</tr>

<!-- week -->
<tr>
  <td>Monday,<br>Feb. 22</td>
  <td>Phylogenetic trees VII: Tree-to-tree comparisons; consensus methods; supertrees (KWW)
  <br><br>
  <a href="lect/lect14.pdf">Lecture Notes</a></td>
  <td><i>Tree Thinking</i>: pp 58-60
  <br><br>
  <a href="https://books.google.com/books?hl=en&lr=&id=ARy6MpHxbw0C&oi=fnd&pg=PA163&dq=A+Classification+of+Consensus+Methods+for+Phylogenetics&ots=RXS2J6XfXK&sig=UFFtswCtc2_g0uEhyNGx2Ei9fEo#v=onepage&q=A%20Classification%20of%20Consensus%20Methods%20for%20Phylogenetics&f=false">
  Bryant, D. (2003). A classification of consensus methods for phylogenetics. DIMACS series in discrete mathematics and theoretical computer science, 61, 163-184.</a></td>
  <td></td>
</tr>

<tr>
  <td>Wednesday,<br>Feb. 24</td>
  <td>Introduction to statistical thinking in phylogenetics (DDA)
  <br><br>
  <a href="lect/lect15.pdf">Lecture Notes</a></td>
  <td>none</td>
  <td><a href="labs/06/lab06.pdf">LAB 06: Bayesian inference using MrBayes and BEAST; Tracer; molecular clocks and fossil calibrations</a></td>
</tr>

<tr>
  <td>Friday,<br>Feb. 26</td>
  <td>Qualitative character evolution within a cladogram I: discrete states; ancestral state reconstructions (DDA)
  <br><br>
  <a href="lect/lect16.pdf">Lecture Notes</a><br>
  <a href="lect/lect16.R">Example R Script</a></td>
  <td><i>Tree Thinking</i>: pp 77-99, 305-322<br><br>
  Maddison WP, Maddison DR (2000) MacClade 4 (pdf manual), Sunderland MA: Sinauer, chapters 3, 4 and 22. <a href="readings/MacClade 4 Manual.pdf">[pdf]</a>
  </td>
  <td></td>
</tr>

<!-- week -->
<tr>
  <td>Monday,<br>Feb. 29</td>
  <td>Qualitative character evolution within a cladogram II: comparing two or more characters (DDA)
  <br><br>
  <a href="lect/lect17.pdf">Lecture Notes</a></td>
  <td><i>Tree Thinking</i>: pp 323-327<br><br>
  Maddison, W.P. 1990. A method for testing the correlated evolution of two binary characters: are gains or losses concentrated on certain branches of a phylogenetic tree? Evolution 44, 539-557.<br><br>
  Donoghue, M. J. 1989. Phylogenies and the analysis of evolutionary sequences, with examples from the seed plants. Evolution 43:1137-1156. (just read to top of p. 1148)<br><br>
  Pagel, M., 1994. Detecting Correlated Evolution on Phylogenies : A General Method for the Comparative Analysis of Discrete Characters. Proceedings R. Soc. Lond B  255:37–45.
  </td>
  <td></td>
</tr>

<tr>
  <td>Wednesday,<br>Mar. 2</td>
  <td>Quantitative character evolution within a cladogram I: intro; ancestral trait reconstruction; phylogenetic    conservatism (DDA)
  <br><br>
  <a href="lect/lect18.pdf">Lecture Notes</a></td>
  <td><i>Tree Thinking</i>: pp 327-336<br><br>
  Felsenstein, J. 1985. Phylogenies and the comparative method. Amer Nat 125:1–15.<br><br>
  Ackerly, D. D. 2009. Conservatism and diversification of plant functional traits: Evolutionary rates versus phylogenetic signal. Proceedings of the National Academy of Sciences USA 106:19699–19706.<br><br>
  Blomberg, S. P., and T. Garland, Jr. 2002. Tempo and mode in evolution: phylogenetic inertia, adaptation and comparative methods. J. Evol. Biol. 15:899-910.</td>
  <td><a href="labs/07/lab07.R">LAB 07: Intro to R and Basic Phylogenetic Functions in R</a></td>
</tr>

<tr>
  <td>Friday,<br>Mar. 4</td>
  <td>Quantitative character evolution within a cladogram II: independent contrasts and trait correlations (DDA)
  <br><br>
  <a href="lect/lect19.pdf">Lecture Notes</a>
  <br>
  <a href="lect/lect19-slides.pdf">Lecture Slides</a></td>
  <td><i>Tree Thinking</i>: pp 336-340<br><br>
  Oakley TH, Cunningham CW. 2000. Independent contrasts succeed where ancestral reconstruction fails in a known bacteriophage phylogeny. Evolution 54:397-405<br><br>
  Garland  Jr, T., P. H. Harvey, and A. R. Ives. 1992. Procedures for the analysis of comparative data using phylogenetically independent contrasts. Systematic Biology 41:18–32.</td>
  <td></td>
</tr>

<!-- week -->
<tr>
  <td>Monday,<br>Mar. 7</td>
  <td>Phylogenetics and adaptation (DDA)
  <br><br>
  <a href="lect/lect20.pdf">Lecture Notes</a></td>
  <td><i>Tree Thinking</i>: pp 77-99, 305-316 (reread)<br><br>
  Donoghue, M. J. 1989. Phylogenies and the analysis of evolutionary sequences, with examples from the seed plants. Evolution 43:1137-1156. (pp. 1148 - end)<br><br>
  Gould SJ, Vrba ES. 1982. Exaptation-a missing term in the science of form. Paleobiology. Jan 1:4-15.</td>
  <td></td>
</tr>

<tr>
  <td>Wednesday,<br>Mar. 9</td>
  <td>Classification I -- introduction to phylogenetic classifications; monophyly, information content (KWW)
  <br><br>
  <a href="lect/lect21.pdf">Lecture Notes</a></td>
  <td><i>Tree Thinking</i>: pp 107-131</td>
  <td><a href="labs/08/lab08.R">LAB 08: Intro to R continued: ancestral state reconstruction, 
  simulating under Brownian motion and Ornstein-Uhlenbeck, and phylogenetically independent contrasts</a></td>
</tr>

<tr>
  <td>Friday,<br>Mar. 11</td>
  <td>Classification II -- phylogenetic taxonomy including incorporation of fossils; Phylocode  (BDM)
  <br><br>
  <a href="lect/lect22.pdf">Lecture Notes</a></td>
  <td><a href="https://www.ohio.edu/phylocode/preface.html">Preface to the Phylocode</a></td>
  <td></td>
</tr>

<!-- week -->
<tr>
  <td>Monday,<br>Mar. 14</td>
  <td>Classification III -- species concepts; speciation (BDM)
  <br><br>
  <a href="lect/lect23.pdf">Lecture Notes</a></td>
  <td><a href="readings/Mishler 2010 (Arp & Ayala).pdf">Mishler, Brent D. 2010. Species are not Uniquely Real Biological Entities. Contemporary Debates in Philosophy of Biology. F. Ayala and R. Arp, Wiley-Blackwell, 110-122</a></td>
  <td></td>
</tr>

<tr>
  <td>Wednesday,<br>Mar. 16</td>
  <td>Classification IV -- DNA barcoding and DNA taxonomy (KWW)
  <br><br>
  <a href="lect/lect24.pdf">Lecture Notes</a></td>
  <td><a href="http://onlinelibrary.wiley.com/enhanced/doi/10.1111/1755-0998.12046/">
  Collins, R. A., & Cruickshank, R. H. (2013). The seven deadly sins of DNA barcoding. Molecular ecology resources, 13(6), 969-975.</a></td>
<td><b>Discuss progress on projects</b><br><br> 
  <a href="labs/09/lab09.pdf">LAB 09: Phylogenetic, Specimen, and Taxonomic Databases and R Packages</a>
  <br><br>
    <b>QUIZ 1 handed out</b> (due Friday evening) - see <a href="quizzes.html">past quizzes</a>
    </td>
</tr>

<tr>
  <td>Friday,<br>Mar. 18</td>
  <td>Classification V -- nomenclature; Zoological & Botanical Codes; practical systematics, monography (KWW) 
  <br><br>
  <a href="lect/lect25.pdf">Lecture Notes</a>
  <br><br>
    <b>QUIZ 1 due tonight!</b>
    </td>
  <td><a href="http://iczn.org/iczn/index.jsp">International Code of Zoological Nomenclature</a> [read preface and skim preamble]
  <br><br>
  <a href="http://www.iapt-taxon.org/nomen/main.php?page=pf">International Code of Nomenclature for algae, fungi, and plants</a> [read preface and skim preamble]
  </td>
  <td></td>
</tr>



<tr>
  <td>Mar. 21 - 25 </td>
  <td colspan=3><i>SPRING BREAK</i> </td>
</tr>



<!-- week -->
<tr>
  <td>Monday,<br>Mar. 28</td>
  <td>Evolution and development - heterochrony  (BDM)</td>
  <td>TBA</td>
  <td></td>
</tr>

<tr>
  <td>Wednesday,<br>Mar. 30</td>
  <td>Molecular evolution  (BDM)</td>
  <td>TBA</td>
  <td>LAB 10: Introduction to RevBayes: phylogenetic analysis using graphical models and Markov chain Monte Carlo </td>
</tr>

<tr>
  <td>Friday,<br>Apr. 1</td>
  <td>Gene family evolution; comparative genomics; evo-devo (BDM)   </td>
  <td>TBA</td>
  <td></td>
</tr>

<!-- week -->
<tr>
  <td>Monday,<br>Apr. 4</td>
  <td>Phylogenetic trees VIII: Below the "species level;" phylogeography; dealing with reticulation (BDM)</td>
  <td>TBA</td>
  <td></td>
</tr>

<tr>
  <td>Wednesday,<br>Apr. 6</td>
  <td>Tempo and mode in macroevolution; patterns of diversification and extinction (BDM)</td>
  <td>TBA</td>
  <td>LAB 11: Coalescence theory: gene tree-species tree reconstruction using RevBayes and the multispecies coalescent</td>
</tr>

<tr>
  <td>Friday,<br>Apr. 8</td>
  <td>Phylogenetics and conservation biology (BDM)</td>
  <td>TBA</td>
  <td></td>
</tr>

<!-- week -->
<tr>
  <td>Monday,<br>Apr. 11</td>
  <td>Comparing sister clades within a cladogram: the shape of evolution  (DDA)</td>
  <td><i>Tree Thinking</i>: pp 370-375<br><br>
  Morlon, H. (2014). Phylogenetic approaches for studying diversification. Ecology Letters, 17(4), 508-525.</td>
  <td></td>
</tr>

<tr>
  <td>Wednesday,<br>Apr. 13</td>
  <td>Adaptive radiations (DDA)</td>
  <td><i>Tree Thinking</i>: pp 375-381<br><br>
  Hodges, S. 1995. Spurring plant diversification: Are floral nectar spurs a key innovation? Proc. Roy. Soc. London Ser. B 262:343-348.<br><br>
  Harmon, L. J., J. A. Schulte, II, A. Larson, and J. B. Losos. 2003. Tempo and mode of evolutionary radiation in Iguanian lizards. Science 301:961-964.<br><br>
  Ackerly, D. D. 2009. Conservatism and diversification of plant functional traits: Evolutionary rates versus phylogenetic signal. Proceedings of the National Academy of Sciences USA 106:19699–19706. (reread)</td>
  <td>LAB 12: <b>discuss progress on projects in class; present initial analysis of project dataset</b>; Birth-death models; 
    joint character evolution and diversification analyses using BiSSE; detecting diversification rate shifts using 
        BAMM
        </td>
</tr>

<tr>
  <td>Friday,<br>Apr. 15</td>
  <td>Phylogenies and Community Ecology I (DDA)</td>
  <td><i>Tree Thinking</i>: pp 359-362<br><br>
  Cavender-Bares, J., D. D. Ackerly, D. Baum, and F. A. Bazzaz. 2004. Phylogenetic overdispersion in Floridian oak communities. Amer. Nat. 163:823-843.<br><br>
  Cavender-Bares, J., K. H. Kozak, P. V. A. Fine, and S. W. Kembel. 2009. The merging of community ecology and phylogenetic biology. Ecology letters 12:693-715.</td>
  <td></td>
</tr>

<!-- week -->
<tr>
  <td>Monday,<br>Apr. 18</td>
  <td>Phylogenies and Community Ecology II (DDA)</td>
  <td>Godoy O, Kraft NJ, Levine JM. 2014. Phylogenetic relatedness and the determinants of competitive outcomes. Ecology letters. Jul 1;17(7):836-44.<br><br>
  Kraft, N.J.B., W.K. Cornwell, C.O. Webb, and D.D. Ackerly. 2007. Trait evolution, community assembly and the phylogenetic structure of ecological communities. American Naturalist 170: 271-283.</td>
  <td></td>
</tr>

<tr>
  <td>Wednesday,<br>Apr. 20</td>
  <td>Biogeography I: basic principles; ecological vs. historical approaches (KWW)</td>
  <td><a href="http://www.sinauer.com/biogeography-4e.html"><i>Biogeography</i>: Chapter 1<a></td>
  <td>LAB 13: Community and spatial phylogenetics: picante, Phylocom; introduction to BIODIVERSE, phylogenetic 
    beta-diversity, mapping
    </td>
</tr>

<tr>
  <td>Friday,<br>Apr. 22</td>
  <td>Biogeography II: vicariance biogeography; detecting dispersal (KWW)</td>
  <td>
  <i>Tree Thinking</i>: pp 349-355<br><br>
  <a href="http://www.sinauer.com/biogeography-4e.html"><i>Biogeography</i>: pp 260-278<a></td>
  <td></td>
</tr>

<!-- week -->
<tr>
  <td>Monday,<br>Apr. 25</td>
  <td>Biogeography III: phylogenetics and range modeling; biome recognition and other spatial issues (BDM)</td>
  <td>TBA</td>
  <td></td>
</tr>

<tr>
  <td>Wednesday,<br>Apr. 27</td>
  <td>Comparing cladograms; cospeciation methods (DDA)</td>
  <td><i>Tree Thinking</i>: pp 355-358<br><br>
  Page, R. D. M. 1996. Temporal congruence revisited: Comparison of mitochondrial DNA sequence divergence in cospeciating pocket gophers and their chewing lice. Systematic Biology 45:151-167. <br><br>
  Percy, D. M., R. D. M. Page, and Q. C. B. Cronk. 2004. Plant-insect interactions: Double-dating associated insect and plant lineages reveals asynchronous radiations. Systematic Biology 53:120- 127. </td>
  <td>LAB 14: <b>discuss progress on projects in class</b>; Probabilistic biogeographic models using 
    BioGeoBEARS and RevBayes
    </td>
</tr>

<tr>
  <td>Friday,<br>Apr. 29</td>
  <td>Coevolution; symbiosis (DDA)<br><br>
  <b>QUIZ 2 handed out</b> (due this evening) - see <a href="quizzes.html">past quizzes</a></td>
  <td>Rezende EL, Lavabre JE, Guimarães PR, Jordano P, Bascompte J. 2007. Non-random coextinctions in phylogenetically structured mutualistic networks. Nature. Aug 23;448(7156):925-8.</td>
  <td></td>
</tr>





<tr>
  <td valign="top">May 2-13 </td>
  <td colspan=3 valign="top"><i>READING &amp; FINALS WEEKS</i></td>
</tr>

<tr>
  <td valign="top">Friday,<br /> May 6</td>
  <td colspan=3 valign="top"><b>Student minisymposium [location and time TBA]</b> </td>
</tr>	

<tr>
  <td valign="top">Wednesday, May 11</td>
  <td colspan=3 valign="top"><b>Final projects due</b> </td>
</tr>	
	
</table>
<br>
<address>
Background image adapted from <a href="https://dx.doi.org/10.1093%2Fbioinformatics%2Fbtl529">a phylogeny of all organisms with complete genome sequences</a>.</i><br>
<br>
Last update Spring 2016 by <a href="http://willfreyman.org">Will Freyman</a><br>
</address>

<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
<script src="assets/bootstrap.min.js"></script>

</body>
</html>
